aries/aries_vcx_core/src/wallet/askar/askar_did_wallet.rs

```diff
@@ -0,0 +1,178 @@
+use aries_askar::{
+    crypto::alg::Chacha20Types,
+    kms::{KeyAlg, LocalKey},
+};
+use async_trait::async_trait;
+use public_key::Key;
+
+use super::{
+    askar_utils::{key_from_base58, local_key_to_bs58_public_key, seed_from_opt},
+    packing::Packing,
+    rng_method::RngMethod,
+    sig_type::SigType,
+    AskarWallet,
+};
+use crate::{
+    errors::error::{AriesVcxCoreError, AriesVcxCoreErrorKind, VcxCoreResult},
+    wallet::{
+        base_wallet::{did_data::DidData, DidWallet},
+        constants::{DID_CATEGORY, TMP_DID_CATEGORY},
+        structs_io::UnpackMessageOutput,
+        utils::did_from_key,
+    },
+};
+
+#[async_trait]
+impl DidWallet for AskarWallet {
+    async fn create_and_store_my_did(
+        &self,
+        seed: Option<&str>,
+        _did_method_name: Option<&str>,
+    ) -> VcxCoreResult<DidData> {
+        let mut tx = self.backend.transaction(self.profile.clone()).await?;
+        let (did, local_key) = self
+            .insert_key(
+                &mut tx,
+                KeyAlg::Ed25519,
+                seed_from_opt(seed).as_bytes(),
+                RngMethod::RandomDet,
+            )
+            .await?;
+        let verkey = local_key_to_bs58_public_key(&local_key)?;
+        self.insert_did(&mut tx, &did, DID_CATEGORY, &verkey, None)
+            .await?;
+        tx.commit().await?;
+        Ok(DidData::new(&did, key_from_base58(&verkey)?))
+    }
+
+    async fn key_for_did(&self, did: &str) -> VcxCoreResult<Key> {
+        let data = self
+            .find_current_did(&mut self.backend.session(self.profile.clone()).await?, did)
+            .await?;
+
+        if let Some(did_data) = data {
+            Ok(did_data.verkey().to_owned())
+        } else {
+            Err(AriesVcxCoreError::from_msg(
+                AriesVcxCoreErrorKind::WalletRecordNotFound,
+                format!("did not found in wallet: {}", did),
+            ))
+        }
+    }
+
+    async fn replace_did_key_start(&self, did: &str, seed: Option<&str>) -> VcxCoreResult<Key> {
+        let mut tx = self.backend.transaction(self.profile.clone()).await?;
+        if self.find_current_did(&mut tx, did).await?.is_some() {
+            let (new_key_name, local_key) = self
+                .insert_key(
+                    &mut tx,
+                    KeyAlg::Ed25519,
+                    seed_from_opt(seed).as_bytes(),
+                    RngMethod::RandomDet,
+                )
+                .await?;
+            let verkey = local_key_to_bs58_public_key(&local_key)?;
+            self.insert_did(&mut tx, did, TMP_DID_CATEGORY, &verkey, None)
+                .await?;
+            tx.commit().await?;
+            key_from_base58(&new_key_name)
```

Comment:
`new_key_name` is first 16 bytes of the public key, so this seems wrong - the function seem to expect key, but gets "key name" instead
Before fixing this, it would be worthy having a look why no test catched this.